ir/src/transition_stmts/graph.rs

```diff
@@ -213,23 +244,46 @@ impl AlgebraicGraph {
         &mut self,
         symbol_table: &SymbolTable,
         vector_access: &VectorAccess,
+        variable_roots: &mut BTreeMap<VariableValue, (TraceSegment, NodeIndex)>,
```

Comment:
nit: We should be able to use the new `VariableRoots` type here. Can you search the crate by `BTreeMap<VariableValue, (TraceSegment, NodeIndex)>` to make sure no other instances were missed?